### PR TITLE
[Espresso] Android Code Inspection and Clean up

### PR DESCRIPTION
**Pull request by hamdikahloun**
## Description

- Handle deprecation & unchecked warning as error

- Avoiding uses or overrides a deprecated API

## Related Issues

flutter/flutter#65970

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by dnfield** [comment]
@adzh - does this get rolled in from somewhere else?

**Comment by adazh** [comment]
> @adzh - does this get rolled in from somewhere else?

Yeah, we initially rolled this out from internal, and there's no automated setup on regular rollout yet.

File: packages/espresso/android/src/main/java/androidx/test/espresso/flutter/action/FlutterViewAction.java [hunk 1]
```diff
@@ -199,6 +199,7 @@ public String getName() {
       return FlutterViewRenderedIdlingResource.class.getSimpleName();
     }
 
+    @SuppressWarnings("deprecation")
```
Comment (xster) [inline packages/espresso/android/src/main/java/androidx/test/espresso/flutter/action/FlutterViewAction.java:202]: put the lint on the specific lines that accesses io.flutter.view.FlutterView

File: packages/espresso/android/src/main/java/androidx/test/espresso/flutter/internal/protocol/impl/DartVmServiceUtil.java [hunk 2]
```diff
@@ -71,6 +71,7 @@ public static String getDartIsolateId(View flutterView) {
   }
 
   /** Gets the Dart executor for the given {@code flutterView}. */
+  @SuppressWarnings("deprecation")
```
Comment (xster) [inline packages/espresso/android/src/main/java/androidx/test/espresso/flutter/internal/protocol/impl/DartVmServiceUtil.java:74]: ditto here

**Review by xster — APPROVED** [review]

File: packages/espresso/android/src/main/java/androidx/test/espresso/flutter/internal/protocol/impl/DartVmServiceUtil.java (hunk 2 above)
Comment (hamdikahloun) [reply packages/espresso/android/src/main/java/androidx/test/espresso/flutter/internal/protocol/impl/DartVmServiceUtil.java]: PR Updated with requested changes . Thank you

**Pull request merged**